app/src/main/java/org/simple/clinic/setup/SetupActivity.kt

```diff
@@ -44,7 +49,12 @@ class SetupActivity : AppCompatActivity() {
   override fun attachBaseContext(baseContext: Context) {
     setupDiGraph()
     val contextWithOverriddenLocale = LocaleOverrideContextWrapper.wrap(baseContext, locale)
-    super.attachBaseContext(ViewPumpContextWrapper.wrap(contextWithOverriddenLocale))
+    val contextWithRouter = wrapContextWithRouter(contextWithOverriddenLocale)
+    val contextWithInjectorProvider = InjectorProviderContextWrapper.wrap(
+        contextWithRouter,
+        mapOf(OnboardingScreenInjector::class.java to component)
+    )
+    super.attachBaseContext(ViewPumpContextWrapper.wrap(contextWithInjectorProvider))
```

Comment:
The result of a wrap is fed into the next call. It's easy to make a mistake here. Can we do something like this?

```kotlin
  override fun attachBaseContext(baseContext: Context) {
    setupDiGraph()
    val wrappedContext = baseContext
        .wrap { LocaleOverrideContextWrapper.wrap(it, locale) }
        .wrap { wrapContextWithRouter(it) }
        .wrap { InjectorProviderContextWrapper.wrap(it, mapOf(OnboardingScreenInjector::class.java to component)) }
        .wrap { ViewPumpContextWrapper.wrap(it) }
    super.attachBaseContext(wrappedContext)
  }

  private fun Context.wrap(wrapper: (Context) -> Context): Context =
      wrapper(this)
```

Reply:
Looks good, I'll do this in #906 since that class has already changed in that PR.

Reply:
Done.